File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/ExtraData.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by akshaykoul on 29/06/16.
 */
public class ExtraData implements Parcelable {
    @SerializedName("value")
    private String value;
    @SerializedName("label")
    private String label;

    public String value() {
        return value;
    }

    public String label() {
        return label;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.value);
        dest.writeString(this.label);
    }

    public ExtraData() {
    }

    protected ExtraData(Parcel in) {
        this.value = in.readString();
        this.label = in.readString();
    }

    public static final Creator<ExtraData> CREATOR = new Creator<ExtraData>
            () {
        @Override
        public ExtraData createFromParcel(Parcel source) {
            return new ExtraData(source);
        }

        @Override
        public ExtraData[] newArray(int size) {
            return new ExtraData[size];
        }
    };

    @Override
    public String toString() {
        return "ExtraData{" +
                "label=" + label +
                "value=" + value +
                '}';
    }
}

File: androidcore/src/main/java/com/example/androidcore/models/story/ExtraData.kt
package com.example.androidcore.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class ExtraData : Parcelable {
    @SerializedName("value")
    private val value: String
    @SerializedName("label")
    private val label: String

    fun value(): String {
        return value
    }

    fun label(): String {
        return label
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.value)
        dest.writeString(this.label)
    }

    protected constructor(`in`: Parcel) {
        this.value = `in`.readString()
        this.label = `in`.readString()
    }

    override fun toString(): String {
        return "ExtraData{" +
                "label=" + label +
                "value=" + value +
                '}'.toString()
    }

    companion object CREATOR : Parcelable.Creator<ExtraData> {
        override fun createFromParcel(parcel: Parcel): ExtraData {
            return ExtraData(parcel)
        }

        override fun newArray(size: Int): Array<ExtraData?> {
            return arrayOfNulls(size)
        }
    }
}